### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by manthanank**
- What are Angular Signals?
- What are Signals?
- Explain Angular Signals with an example
- What are the Route Parameters? Could you explain each of them?

The above questions were added with answers.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4719,3 +4724,92 @@
       ```
       
       **[⬆ Back to Top](#table-of-contents)**
+
+278. ### What are Angular Signals?
```
Comment (jdegand) [inline README.md:4728]: You duplicated this question twice.

File: README.md [hunk 2]
```diff
@@ -4719,3 +4724,92 @@
...
+      A signal is a wrapper around a value that can notify interested consumers when that value changes. Signals can contain any value, from simple primitives to complex data structures.
+
+      **[⬆ Back to Top](#table-of-contents)**
+      
+279. ### Explain Angular Signals with an example. 
+      In this example, we create a signal named `count` and initialize it with a value of 0. We then connect to the signal, allowing us to be notified whenever its value changes. Finally, we add a button that increments the count when clicked.
+
+      When the button is clicked, the `incrementCount()` method is called. This method sets the new value of the `count` signal to 1. Objects connected to the signal (subscribers) are then notified of the change, and the updated value is displayed in the UI.
+
+      In typescript file
```
Comment (jdegand) [inline README.md:4743]: `TypeScript`

File: README.md [hunk 3]
```diff
@@ -4719,3 +4724,92 @@
...
       
       **[⬆ Back to Top](#table-of-contents)**
+
+278. ### What are Angular Signals?
+      A signal is a wrapper around a value that can notify interested consumers when that value changes. Signals can contain any value, from simple primitives to complex data structures.
+
+      **[⬆ Back to Top](#table-of-contents)**
+
+278. ### What are Signals?
+      A signal is a wrapper around a value that can notify interested consumers when that value changes. Signals can contain any value, from simple primitives to complex data structures.
```
Comment (jdegand) [inline README.md:4734]: when `the` value changes

File: README.md [hunk 4]
```diff
@@ -4719,3 +4724,92 @@
...
+
+      @Component({
+        selector: 'my-app',
+        templateUrl: './app.component.html',
+        styleUrls: ['./app.component.css']
+      })
+      export class AppComponent implements OnInit {
+
+        count = signal(0); // Initial value of count
+        displayedCount = computed(() => this.count()); // Computed signal for display
```
Comment (jdegand) [inline README.md:4757]: You are not doing anything inside the `computed`.  Change to the typical example of:

doubleCount = computed(()=> this.count() * 2)

File: README.md [hunk 5]
```diff
@@ -4719,3 +4724,92 @@
...
+
+        ngOnInit() {
+          // Subscribe to displayedCount instead of count for updates in the template
+          this.displayedCount.subscribe(value => {
+            console.log('Displayed count changed to', value); // Optional logging for debugging
+          });
+        }
+
+        incrementCount() {
+          this.count.set(this.count() + 1); // Update count using set() for immutability
```
Comment (jdegand) [inline README.md:4769]: You wouldn't use `set` here.  You would use `update`.   Delete the comment as well.

File: README.md [hunk 6]
```diff
@@ -4719,3 +4724,92 @@
...
+        }
+      }
+      ```
+      In HTML file
+      ```html
+      <h1>Angular Signals Example</h1>
+
+      <button (click)="incrementCount()">Increment Count</button>
+
+      <p>Count: {{ count }}</p>
```
Comment (jdegand) [inline README.md:4779]: Needs to be `count()`.

**Comment by sudheerj** [comment]
@manthanank  Thanks for PR. Can you remove stackblitz link? It is not required.

**Pull request merged**